pkg/controller/controller_broker.go

```diff
@@ -224,15 +229,35 @@ func (c *controller) reconcileServiceBroker(broker *v1alpha1.ServiceBroker) erro
 		}
 		glog.V(5).Infof("Successfully converted catalog payload from ServiceBroker %v to service-catalog API", broker.Name)
 
-		if len(catalog) == 0 {
+		if len(serviceClasses) == 0 {
 			s := fmt.Sprintf("Error getting catalog payload for broker %q; received zero services; at least one service is required", broker.Name)
 			glog.Warning(s)
 			c.recorder.Eventf(broker, api.EventTypeWarning, errorSyncingCatalogReason, s)
 			c.updateServiceBrokerCondition(broker, v1alpha1.ServiceBrokerConditionReady, v1alpha1.ConditionFalse, errorSyncingCatalogReason, errorSyncingCatalogMessage+s)
 			return stderrors.New(s)
 		}
 
-		for _, serviceClass := range catalog {
+		for _, servicePlan := range servicePlans {
```

Comment:
nit: is there a reason you processed the plans first? my intuition would be to process the services before the plans.

Reply:
admission controller might see things in the wrong order. Still might, in fact.